src/Diag/Result.hs

```diff
@@ -171,15 +172,29 @@ renderFailure ws (ErrGroup _ ectx es) headerDoc = header headerDoc <> renderedCt
     renderedErrs :: Doc AnsiStyle
     renderedErrs =
       section "Relevant errors" $
-        subsection "Error" (map renderErrWithStack (NE.toList es))
+        subsection "Error" (map (`renderErrWithStack` True) (NE.toList es))
 
     renderedPossibleErrs :: Doc AnsiStyle
     renderedPossibleErrs =
       case ws of
         [] -> emptyDoc
         _ ->
           section "Possibly-related warnings" $
-            subsection "Warning" (map renderEmittedWarn ws)
+            unannotatedSubsection (map (renderEmittedWarn True) ws)
+
+renderFailureWithoutWarnings :: ErrGroup -> Doc AnsiStyle -> Doc AnsiStyle
+renderFailureWithoutWarnings (ErrGroup _ ectx es) headerDoc = header headerDoc <> renderedCtx <> renderedErrs
+  where
+    renderedCtx :: Doc AnsiStyle
```

Comment:
Since `renderedCtx` and `renderedErr` are essentially the same as the definitions in `renderFailure` could you break them out into top-level functions that can be shared between the two?